probe/endpoint/dns_snooper_linux_amd64.go

```diff
@@ -203,6 +208,25 @@ func (s *DNSSnooper) run() {
 	}
 }
 
+// handleDecodeError logs errors up to the maximum allowed count
+func (s *DNSSnooper) handleDecodingError(err error) {
+	// prevent potential memory leak
+	if len(s.decodingErrorCounts) > maxDecodingErrorCardinality {
```

Comment:
optional: maybe have a flag for "reached max cardinality", so the first time we reach it we can emit a final `DNSSnooper: Too many unique error messages, silencing all errors` or similar?